postcard_creator/postcard_img_util.py

```diff
@@ -0,0 +1,174 @@
+            logger.info('exporting image to {} (image_export=True)'.format(path))
+            cover.save(path)
+
+    return scaled
+
+
+def create_text_image(text, image_export=False, **kwargs):
+    """
+    Create a jpg with given text and return in bytes format
+    """
+    text_canvas_w = 720
+    text_canvas_h = 744
+    text_canvas_bg = 'white'
+    text_canvas_fg = 'black'
+    text_canvas_font_name = 'open_sans_emoji.ttf'
+
+    def load_font(size):
+        return ImageFont.truetype(pkg_resources.resource_stream(__name__, text_canvas_font_name), size)
+
+    def find_optimal_size(msg, min_size=20, max_size=400, min_line_w=1, max_line_w=80, padding=0):
+        """
+        Find optimal font size and line width for a given text
+        """
+
+        def line_width(font_size, line_padding=70):
+            l = min_line_w
+            r = max_line_w
+            font = load_font(font_size)
+            while l < r:
+                n = floor((l + r) / 2)
+                t = ''.join([char * n for char in '1'])
+                font_w, font_h = font.getsize(t)
+                font_w = font_w + (2 * line_padding)
+                if font_w >= text_canvas_w:
+                    r = n - 1
+                    pass
+                else:
+                    l = n + 1
+                    pass
+            return n
+
+        size_l = min_size
+        size_r = max_size
+        last_line_w = 0
+        last_size = 0
+
+        while size_l < size_r:
+            size = floor((size_l + size_r) / 2.0)
+            last_size = size
+
+            line_w = line_width(size)
+            last_line_w = line_w
+
+            lines = textwrap.wrap(msg, width=line_w)
+            font = load_font(size)
+            total_w, line_h = font.getsize(msg)
+            tot_height = len(lines) * line_h
+
+            if tot_height + (2 * padding) < text_canvas_h:
+                start_y = (text_canvas_h - tot_height) / 2
+            else:
+                start_y = 0
+
+            if start_y == 0:
+                size_r = size - 1
+            else:
+                # does fit
+                size_l = size + 1
+
+        return last_size, last_line_w
+
+    def center_y(lines, font_h):
+        tot_height = len(lines) * font_h
+        if tot_height < text_canvas_h:
+            return (text_canvas_h - tot_height) / 2
+        else:
+            return 0
+
+    size, line_w = find_optimal_size(text, padding=50)
+    logger.debug(f'using font with size: {size}, width: {line_w}')
+
+    font = load_font(size)
+    font_w, font_h = font.getsize(text)
+    lines = textwrap.wrap(text, width=line_w)
+    text_y_start = center_y(lines, font_h)
+
+    canvas = Image.new('RGB', (text_canvas_w, text_canvas_h), text_canvas_bg)
+    draw = ImageDraw.Draw(canvas)
+    for line in lines:
+        width, height = font.getsize(line)
+        draw.text(((text_canvas_w - width) / 2, text_y_start), line,
```

Comment:
Depending on the image resolution I get the following error message:

```
 File "/home/pi/postCards/postcard_creator/postcard_creator.py", line 96, in wrapped
    return func(*args, **kwargs)
  File "/home/pi/postCards/postcard_creator/postcard_creator_swissid.py", line 117, in send_free_card
    img_text_base64 = base64.b64encode(self.create_text_cover(postcard.message)).decode('ascii')
  File "/home/pi/postCards/postcard_creator/postcard_creator_swissid.py", line 153, in create_text_cover
    return create_text_image(msg, image_export=True)
  File "/home/pi/postCards/postcard_creator/postcard_img_util.py", line 163, in create_text_image
    embedded_color=True)
  File "/home/pi/.local/lib/python3.7/site-packages/PIL/ImageDraw.py", line 379, in text
    draw_text(ink)
  File "/home/pi/.local/lib/python3.7/site-packages/PIL/ImageDraw.py", line 361, in draw_text
    self.im.paste(color, coord + coord2, mask)
TypeError: integer argument expected, got float
```

I suggest to use an interger division here:

```
draw.text(((text_canvas_w - width) // 2, text_y_start), line,
```

Reply:
nice catch! Yes, LGTM